### ISSUE TITLE
docker start hangs in 20.10.4 but not 20.10.3

### ISSUE THREAD

**Issue by akomakom** [moby/moby#42101]
**swarm container hangs on start in 20.10.4 but not 20.10.3**

After a recent upgrade (see below), starting a container (**swarm** in my case) hangs on `start`, and a `kill -9` is the only thing that stops it.
```
Feb 28 05:01:13 Updated: 1:docker-ce-cli-20.10.4-3.el7.x86_64
Feb 28 05:01:17 Updated: 3:docker-ce-20.10.4-3.el7.x86_64
Feb 28 05:01:18 Updated: docker-ce-rootless-extras-20.10.4-3.el7.x86_64
```

**Steps to reproduce the issue:**
1. On a CentOS 7 system, install docker 20.10.4.  (I cleaned out data/run dirs, it makes no difference)
2. Pull: `docker pull swarm`
3. Create a container: `/usr/bin/docker create --net host -m 0b -l protected  --name SwarmManager swarm  manage --strategy spread -H tcp://0.0.0.0:2375 etcd://127.0.0.1:2379`
4. Start Container: `docker start -a SwarmManager` (no output, hangs, does not respond to Ctrl-C)

*With 20.10.3 the above produces output and responds to Ctrl-C*

**Describe the results you received:**

`docker start` hangs, produces no output and does not respond to Ctrl-C or `kill`.  Only `kill -9` on the docker process terminates the command.

**Describe the results you expected:**
```
[ root@host ] docker start -a SwarmManager
time="2021-03-01T16:58:22Z" level=info msg="Initializing discovery without TLS" 
time="2021-03-01T16:58:22Z" level=info msg="Listening for HTTP" addr="0.0.0.0:2375" proto=tcp 
time="2021-03-01T16:58:22Z" level=info msg="Registered Engine xxxxxxxx at xxxxxxx:4243" 
...
```
*should respond to Ctrl-C*

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**
```
Client: Docker Engine - Community
 Version:           20.10.4
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        d3cb89e
 Built:             Thu Feb 25 07:06:20 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.4
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Thu Feb 25 07:04:45 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker info`:**

```
Client:                       
 Context:    default
 Debug Mode: false                    
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
                                            
Server:                  
 Containers: 1                          
  Running: 0 
  Paused: 0                 
  Stopped: 1
 Images: 1
 Server Version: 20.10.4
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
WARNING: API is accessible on http://0.0.0.0:4243 without encryption.
         Access to the remote API is equivalent to root access on the host. Refer
         to the 'Docker daemon attack surface' section in the documentation for
         more information: https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc version: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-1160.15.2.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 11.58GiB
 Name: xxxxx-swarm-01.xxxx.xxxx.xxxx
 ID: 7UO7:7HWW:XWIV:XJO6:NR2V:HELO:PBHW:WKJP:2YRV:INZY:EP4N:KA6Z
 Docker Root Dir: /data/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  xxxxxxxx03.xxxx.xxxx.xxxx:4443
  dtr.xxxx.xxxx.xxxx:4443
  xxxx-swarm-01.xxxx.xxxx.xxxx:5000
  127.0.0.0/8
 Live Restore Enabled: false

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

* CentOS Linux release 7.9.2009 (Core)
* Using default file driver (this machine only runs the swarm manager)
* Dockerd: `/usr/bin/dockerd --data-root /data/docker -H tcp://0.0.0.0:4243 -H unix:///var/run/docker.sock --ip-forward=true --iptables=true --ip-masq=true --ipv6 --fixed-cidr-v6 2001:db8:1::/64 --log-driver json-file --log-opt max-size=10m --log-opt max-file=3 -G docker  --mtu=1600`

**Comment by iwharris** [comment]
We're experiencing the same behavior on 20.10.4 - downgrading to 20.10.3 resolves the issue currently.

We've reproduced it on Ubuntu 18.04.5 and Ubuntu 20.04.2

These repro steps hang on 20.10.4 but work on 20.10.3:

```bash
docker create --rm --name test nginx:1.19-alpine
docker start test -a
```

**Comment by bkuebler** [comment]
Same issue with docker create and docker start -a here after upgrading to 20.10.4. Prior release works.

**Comment by ghomem** [comment]
Not sure if this is related but on Ubuntu 20.04:

* 5:20.10.4~3-0~ubuntu-focal is broken
* 5:20.10.3~3-0~ubuntu-focal works

I was running some tests yesterday with puppetlabs-docker and had to pin the docker version to 5:20.10.3~3-0~ubuntu-focal.

**Comment by thaJeztah** [comment]
thanks for reporting; this is a duplicate of https://github.com/moby/moby/issues/42093, and fixed in docker cli 20.10.5 through https://github.com/docker/cli/pull/2987
